catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/MetacardFactory.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.impl.operations;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardCreationException;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.impl.FrameworkProperties;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.security.Subject;
+import ddf.security.SubjectUtils;
+
+/**
+ * Support class for creating metacards for the {@code CatalogFrameworkImpl}.
+ *
+ * This factory class contains methods specific to metacard creation for the CFI and its
+ * support classes. No operations/support methods should be added to this class except in support
+ * of CFI metacard creation.
+ */
+public class MetacardFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetacardFactory.class);
+
+    //
+    // Injected properties
+    //
+    private final FrameworkProperties frameworkProperties;
```

Comment:
Looks like this class only needs to call `getMimeTypeToTransformerMapper()` on `frameworkProperties`. Should we only inject that instead of the whole `FrameworkProperties` monster object?